dash/scripts/dash-test.sh

```diff
@@ -85,7 +85,7 @@ run_suite()
   # Number of failed tests in this run
   THIS_FAIL_COUNT=$(($NEW_FAIL_COUNT-$TOTAL_FAIL_COUNT))
   TOTAL_FAIL_COUNT=$NEW_FAIL_COUNT
-  if [ "$THIS_FAIL_COUNT" = "0" ]; then
+  if [ $TEST_RET -a "$THIS_FAIL_COUNT" = "0" ]; then
```

Comment:
While `-a` is POSIX, it is not portable to all shell implementations. In autoconf world we avoid it tough. I personally think that my eye are more trained to '&&': `if  [ $TEST_RET ] && [ "$THIS_FAIL_COUNT" = "0" ]; then`. Though now that `$TEST_RET` stands alone, it is also clear that you does not test what you expect. `TEST_RET` is always a non-empty string. And this is what you test for, i.e., it is always true. That should be `[ $TEST_RET -eq 0 ]`. And while you are at it, why not use an arithmetic test for the latter part too: `[ $THIS_FAIL_COUNT -eq 0 ]`.

Reply:
Also, when you catch the return code in line 79m you get not the one from the actual test but for the last command in pipeline, which is `grep -v '^#'`.

Reply:
Thanks for looking at this and for the hints, they should be addressed in my recent changes.